lib/oelite/function.py

```diff
@@ -128,32 +128,43 @@ def __call__(self):
 class ShellFunction(OEliteFunction):
 
     def __init__(self, meta, var, name=None, tmpdir=None):
+        self.result = None
         super(ShellFunction, self).__init__(meta, var, name, tmpdir)
         return
 
-    def runscript(self, cmd):
-        cmdstr = cmd
+    def wait(self, poll):
+        if self.result is not None:
+            return self.result
+        if poll:
+            ret = self.subprocess.poll()
+        else:
+            ret = self.subprocess.wait()
+        if ret is None:
+            return None
+        if ret == 0:
+            self.result = True
+        else:
+            self.result = False
+            print "Error: Command failed: %r: %d"%(self.cmdstr, ret)
+        return self.result
+
+
+    def startscript(self, cmd):
```

Comment:
See above. Previously, runscript made sense because we were still doing everything serially. Now we just want to start it, so I changed the name.